src/notifications/ManagerLinux.cpp

```diff
@@ -167,6 +182,68 @@ NotificationsManager::notificationClosed(uint id, uint reason)
         notificationIds.remove(id);
 }
 
+/**
+ * @param text This should be an HTML-formatted string.
+ *
+ * If D-Bus says that notifications can have body markup, this function will
+ * automatically format the notification to follow the supported HTML subset
+ * specified at https://www.freedesktop.org/wiki/Specifications/StatusNotifierItem/Markup/
+ */
+QString
+NotificationsManager::formatNotification(const mtx::responses::Notification &notification)
+{
+        const auto sender =
+          cache::displayName(QString::fromStdString(notification.room_id),
+                             QString::fromStdString(mtx::accessors::sender(notification.event)));
+
+        // TODO: decrypt this message if the decryption setting is on in the UserSettings
+        if (auto msg = std::get_if<mtx::events::EncryptedEvent<mtx::events::msg::Encrypted>>(
+              &notification.event);
+            msg != nullptr)
+                return tr("%1 sent an encrypted message").arg(sender);
+
+        const auto messageLeadIn =
+          ((mtx::accessors::msg_type(notification.event) == mtx::events::MessageType::Emote)
+             ? "* " + sender + " "
+             : sender +
+                 (utils::isReply(notification.event)
+                    ? tr(" replied",
+                         "Used to denote that this message is a reply to another "
+                         "message. Displayed as 'foo replied: message'.")
+                    : "") +
+                 ": ");
+
+        if (hasMarkup_) {
+                if (hasImages_ && mtx::accessors::msg_type(notification.event) ==
+                                    mtx::events::MessageType::Image) {
+                        QString imgPath = cacheImage(notification.event);
+                        if (imgPath.isNull())
+                                return mtx::accessors::formattedBodyWithFallback(notification.event)
+                                  .prepend(messageLeadIn);
+                        else
+                                return QString("<img src=\"file:///" + imgPath + "\" alt=\"" +
+                                               mtx::accessors::formattedBodyWithFallback(
+                                                 notification.event) +
+                                               "\">")
+                                  .prepend(messageLeadIn);
+                }
+
+                return mtx::accessors::formattedBodyWithFallback(notification.event)
+                  .prepend(messageLeadIn)
+                  .replace("<em>", "<i>")
+                  .replace("</em>", "</i>")
+                  .replace("<strong>", "<b>")
+                  .replace("</strong>", "</b>")
+                  .replace(QRegularExpression("(<mx-reply>.+\\<\\/mx-reply\\>)"), "");
```

Comment:
You should put this regex somewhere reusable. Also, why the backslashes?

Reply:
Probably because I came up with it not really knowing how regexes work and forgot to change it

Reply:
Just make the one from the timeline available everywhere as a constant :3